### PR TITLE
[AMDGCN] Use ZExt when handling indices in insertment element 

### LINKED ISSUE #85717: [AMDGPU] Uses SExt on indices in `isInterestingPHIIncomingValue`
## Reproduction

https://godbolt.org/z/P97TT7fEs

## Cause

When `i1 true` is used as an index, SExt extends it to `i32 -1`. This would cause BitVector to overflow.
This can be a quick fix by using ZExt over SExt, as we have specified in the language manual that the index shall be treated as an unsigned number. (https://llvm.org/docs/LangRef.html#insertelement-instruction)

### PR DESCRIPTION
**Pull request by DataCorrupted**
When i1 true is used as an index, SExt extends it to i32 -1. This would cause BitVector to overflow.
The language manual have specified that the index shall be treated as an unsigned number, this patch fixes that. (https://llvm.org/docs/LangRef.html#insertelement-instruction)

This patch fixes #85717

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/pr85717.ll [hunk 1]
```diff
@@ -0,0 +1,24 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+;RUN: llc -mtriple=amdgcn -mcpu=gfx1036 < %s | FileCheck %s
+define void @test(i1 %Bool, ptr %Ptr, <32 x float> %Vec1, <32 x float> %Vec2) {
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/pr85717.ll:3]: Name test  pr85717? Can this merge in with the existing test for this?

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/AMDGPU/amdgpu-codegenprepare-break-large-phis.ll [hunk 2]
```diff
@@ -1197,3 +1197,54 @@ reallyfinally:
   store <5 x double> %val, ptr %out, align 1
   ret void
 }
+
+define amdgpu_kernel void @zext_i1_as_index(i1 %Bool, ptr %Ptr, <4 x float> %Vec1, <4 x float> %Vec2) {
+; OPT-LABEL: @test(
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/amdgpu-codegenprepare-break-large-phis.ll:1202]: Label name doesn't match. Also use the issue number

Comment (DataCorrupted) [reply llvm/test/CodeGen/AMDGPU/amdgpu-codegenprepare-break-large-phis.ll]: Done

**Comment by DataCorrupted** [comment]
Kindly let me know if there are anything else.

**Review by arsenm — APPROVED** [review]

**Pull request merged**